service/matching/db_task_queue_ownership.go

```diff
@@ -116,13 +130,21 @@ func (m *dbTaskQueueOwnershipImpl) getAckedTaskID() int64 {
 	m.Lock()
 	defer m.Unlock()
 
+	if m.status == dbTaskQueueOwnershipStatusLost {
+		return -1
```

Comment:
will caller panic with this value?

Reply:
No. This only ensures that no task will be considered as ACKed.